process_report/tests/unit_tests.py

```diff
@@ -412,3 +415,36 @@ def test_apply_credit_0002(self):
                 ["OpenShift GPUA100SXM4", "OpenStack GPUA100SXM4"],
             )
             self.assertEqual(row["Charge"], row["SU Charge"] * row["SU Hours"])
+
+
+class TestUploadToS3(TestCase):
+    @mock.patch("process_report.process_report.get_invoice_bucket")
+    @mock.patch("process_report.process_report.get_iso8601_time")
+    def test_remove_prefix(self, mock_get_time, mock_get_bucket):
+        mock_bucket = mock.MagicMock()
+        mock_get_bucket.return_value = mock_bucket
+        mock_get_time.return_value = "0"
+
+        invoice_month = "2024-03"
+        filenames = ["test.csv", "test2.test.csv", "test3"]
+        answers = [
+            ("test.csv", f"Invoices/{invoice_month}/test {invoice_month}.csv"),
+            (
+                "test.csv",
+                f"Invoices/{invoice_month}/Archive/test {invoice_month} 0.csv",
+            ),
+            (
+                "test2.test.csv",
+                f"Invoices/{invoice_month}/test2.test {invoice_month}.csv",
+            ),
+            (
+                "test2.test.csv",
+                f"Invoices/{invoice_month}/Archive/test2.test {invoice_month} 0.csv",
+            ),
+            ("test3", f"Invoices/{invoice_month}/test3 {invoice_month}.csv"),
+            ("test3", f"Invoices/{invoice_month}/Archive/test3 {invoice_month} 0.csv"),
+        ]
+
+        process_report.upload_to_s3(filenames, invoice_month)
+        for i, call_args in enumerate(mock_bucket.upload_file.call_args_list):
```

Comment:
You don't need to iterate here as you can compare the lists directly with `assertEqual`. https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args_list

```python
calls = [
            (("test.csv", f"Invoices/{invoice_month}/test {invoice_month}.csv"),),
            (("test.csv", f"Invoices/{invoice_month}/Archive/test {invoice_month} 0.csv"),),
            (("test2.test.csv", f"Invoices/{invoice_month}/test2.test {invoice_month}.csv"),),
            (("test2.test.csv", f"Invoices/{invoice_month}/Archive/test2.test {invoice_month} 0.csv"),),
            (("test3", f"Invoices/{invoice_month}/test3 {invoice_month}.csv"),),
            (("test3", f"Invoices/{invoice_month}/Archive/test3 {invoice_month} 0.csv"),)
        ]

        process_report.upload_to_s3(filenames, invoice_month)
        self.assertEqual(calls, mock_bucket.upload_file.call_args_list)
```

But this is fine too.

Reply:
Meaning, I don't need to make any changes for now?

Reply:
Yeah, just pointing out a possible improvement for your future reference.